logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -163,25 +167,47 @@ internal class CallDataSource(
     private val scope = CoroutineScope(job + kaliumDispatchers.io)
     private val callJobs = ConcurrentMutableMap<ConversationId, Job>()
     private val staleParticipantJobs = ConcurrentMutableMap<QualifiedClientID, Job>()
+    private val _recentlyEndedCallFlow = MutableSharedFlow<RecentlyEndedCall>(
+        extraBufferCapacity = 1
+    )
 
     override suspend fun observeCurrentCall(conversationId: ConversationId): Flow<Call?> = _callMetadataProfile.map {
-        it[conversationId]?.let { currentCall ->
-            Call(
-                conversationId = conversationId,
-                status = currentCall.callStatus,
-                isMuted = currentCall.isMuted,
-                isCameraOn = currentCall.isCameraOn,
-                isCbrEnabled = currentCall.isCbrEnabled,
-                callerId = currentCall.callerId,
-                conversationName = currentCall.conversationName,
-                conversationType = currentCall.conversationType,
-                callerName = currentCall.callerName,
-                callerTeamName = currentCall.callerTeamName,
-                establishedTime = currentCall.establishedTime,
-                participants = currentCall.getFullParticipants(),
-                maxParticipants = currentCall.maxParticipants
+        it[conversationId]?.mapCallMetadataToCall(conversationId)
+    }
+
+    override suspend fun updateRecentlyEndedCall(
+        conversationId: ConversationId,
+        callMetadata: CallMetadata,
+        callEndReason: Int
+    ) {
+        _recentlyEndedCallFlow.tryEmit(
```

Comment:
please also check https://github.com/wireapp/kalium/pull/3113 as this changes this solution to a more compact which does not expose that much